src/logging/monitor.ts

```diff
@@ -51,37 +56,75 @@ async function* splitStream(chunkIterable: AsyncIterable<any>) {
 	}
 }
 
+const getSupervisorContainer =
+	async (): Promise<ContainerInspectInfo | null> => {
+		let container: ContainerInspectInfo | null = null;
+		try {
+			container = await docker.getContainer('balena_supervisor').inspect();
+		} catch {
+			// Ignore errors getting the supervisor container
+		}
+
+		try {
+			container = await docker.getContainer('resin_supervisor').inspect();
+		} catch {
+			// Ignore errors getting the legacy supervisor container
+		}
+
+		return container;
+	};
+
+const isSupervisorLog = (row: JournalRow) =>
+	row.CONTAINER_NAME === 'balena_supervisor' ||
+	row.CONTAINER_NAME === 'resin_supervisor';
+
+const isSupervisorStartupMessage = (row: JournalRow) => {
+	if (!isSupervisorLog(row)) {
+		return false;
+	}
+	const message = messageFieldToString(row.MESSAGE);
+	if (message == null) {
+		return false;
+	}
+	return /Supervisor v.+ starting up/.test(message);
+};
+
 /**
  * Streams logs from journalctl and calls container hooks when a record is received matching container id
  */
 class LogMonitor {
+	// Additional host services we want to stream the logs for
+	private HOST_SERVICES = [
+		'os-power-mode.service',
+		'os-fan-profile.service',
+		'nvpmodel.service',
+	];
+
 	private containers: {
 		[containerId: string]: {
 			hook: MonitorHook;
 		};
 	} = {};
 	private setupAttempts = 0;
 
-	// Only stream logs since the start of the supervisor
+	// By default, only stream logs since the start of the Supervisor process
 	private lastSentTimestamp = Date.now() - performance.now();
 
-	public async start(): Promise<void> {
+	public async start(logSystemMessage: LogSystemMessage): Promise<void> {
```

Comment:
I don't understand this new interface. The `start` function receives a function type that it calls if it sees the "Supervisor starting up" message. The `logSystemMessage` function then calls `log` on this same object.

I don't think the interface is very clean, it gives additional responsibilities to the monitor interface and introduces circular dependencies.

Reply:
Ok, I understand why we need to pass the `logSystemMessage` to monitor. Is because there are only container hooks, but not a `system` hook. 

I propose these changes
- Reuse the MonitorHook interface as the system hook, do not use the `LogSystemMessage` type, that function signature is very hard to understand and we should change it anyway. This gets rid of the circular dependency
- Do not pass the hook on start. Add an `attachSystemLogger` (or something like that) to the object interface. That mimics what already exists for container hook. Passing the hook on the `start` function does not make a lot of sense

I still don't like that this module now also has to report supervisor start. I would try first to get to the bottom of why we have issues with the out of order messages. If that's indeed an issue, then perhaps we could defer system messages to after start or skip the "supervisor" start message altogether if there are previous logs.

I also don't like depending on the "Supervisor starting up" message being on the logs as it creates a hidden interface

Reply:
I ran some additional tests after making some changes to the code per your review @pipex , and found that log-streaming of timestamps earlier than the `Supervisor starting` log line is now working. The changes I made were to `attachSystemHook` within the `initialized` method of `src/logger.ts` (now merged into `src/logging/index.ts`, henceforth referenced as Logger), and to use the generic `log` method provided by the Logger instead of `logSystemMessage`. I gather that this is what made a difference, however I am only certain in the fact that the log-streaming works now, and not much else.

Therefore, the changes no longer depend on `Supervisor starting` log line being in the logs and no longer have to parse that. The hook is also not passed on `logMonitor.start`, instead passed with logMonitor.attachSystemHook` which is called within `logger.initialized`. This simplifies the code from the previous iteration.